src/test/java/org/opensearch/security/transport/SecurityInterceptorTests.java

```diff
@@ -201,17 +191,48 @@ public <T extends TransportResponse> void sendRequest(
         assertEquals(threadPool.getThreadContext().getHeader(ConfigConstants.OPENDISTRO_SECURITY_USER_HEADER), null);
     }
 
+
+    /**
+     * Tests the scenario when remote node is on same OS version
```

Comment:
If I'm interpreting this correctly, all versions will use JDK serialization (except 2.11.0)?
Can we add a test to verify what should happen if shouldUseJDKSerialization is set to false when version is current for both?